src/Library/demos/Account/code.rs

```diff
@@ -0,0 +1,45 @@
+use std::error::Error;
+
+use crate::workbench;
+use ashpd::desktop::account::UserInformation;
+use glib::MainContext;
+use gtk::prelude::*;
+use gtk::{gdk, gio, glib};
+
+pub fn main() {
+    let button: gtk::Button = workbench::builder().object("button").unwrap();
+    button.connect_clicked(|_| {
+        MainContext::default().spawn_local(async {
+            if let Err(err) = request_user_information().await {
+                eprintln!("Could not request user information: {err}")
+            }
+        });
+    });
+}
+
+async fn request_user_information() -> Result<(), Box<dyn Error>> {
+    let entry: adw::EntryRow = workbench::builder().object("entry").unwrap();
+
+    let response = UserInformation::request()
+        .reason(entry.text().as_str())
```

Comment:
Do you actually have to call as_str btw?

Reply:
also i am sad you don't use window identifier :P

Reply:
> Do you actually have to call as_str btw?

Yes, that is needed

> also i am sad you don't use window identifier :P

Good point, fixed